desc/compute/geom_utils.py

```diff
@@ -37,11 +37,14 @@ def rotation_matrix(axis, angle=None):
     -------
     rot : ndarray, shape(3,3)
         Matrix to rotate points in cartesian (X,Y,Z) coordinates
+
     """
+    # FIXME: does not work with AD
```

Comment:
You could probably add a cond here that checks if axis is all zero (or otherwise small) and just returns the identity matrix in that case? I think that might solve the AD issue

Reply:
OK I got that to work, but I had to get rid of the optional `angle` argument and make it always use the norm of `axis` as the angle. The issue is that the derivative of `jax.linalg.norm(x)` is NaN when `x = 0`, in both forward and reverse mode. So we can't call `norm(axis)` outside of the `cond` branch where it is guaranteed to be nonzero.

Reply:
I think this is what you want: https://github.com/PlasmaControl/DESC/blob/c837cc544aaaafacec2f54c9f2ccf32156ba6b4c/desc/compute/utils.py#L492-L493

Reply:
If we can use `safediv` and keep the angle argument, I prefer that as it is much clearer than having the angle be the norm of the axis

Reply:
I think I have this working now with the angle argument. I also created a new util function `safenormalize` to normalize vectors to unit length